File: CodingTest/성호/20주차/숫자게임.js
function solution(A, B) {
  let answer = 0;
  let visited = Array(A.length).fill(false);
  A.sort((a, b) => b - a);
  B.sort((a, b) => b - a);
  let left = 0,
    right = 0;
  while (left < A.length) {
    if (A[left] < B[right]) {
      right++;
      answer++;
    }
    left++;
  }
  return answer;
}

File: CodingTest/성유진/21주차/숫자게임.java
import java.util.*;

class Solution {
    public int solution(int[] A, int[] B) {
        int N = A.length;
        List<Integer> listB = new ArrayList<>();
        Arrays.sort(B);
        for (int i = 0; i < N; i++) {
            listB.add(B[i]);
        }
        
        int answer = 0;
        for (int i = 0; i < N; i++) {
            int idx = Collections.binarySearch(listB, A[i] + 1);
            if (idx < 0) {
                idx = -idx - 1;
            }
            if (idx < listB.size()) {
                listB.remove(idx);
                answer++;   
            }
        }
        return answer;
    }
}
